src/main/java/com/communityratesgames/transactions/RatingService.java

```diff
@@ -49,44 +49,46 @@ public Rating findByGameIdAndUserId(String gameId, String userId) {
-        String title = rating.getGame();
         try {
-            if (findByGameIdAndUserId(title, rating.getUser()) == null) {
+            rating.setCreationDate(Timestamp.from(Instant.now()));
+            Rating newRating = ratingModelToEntity(rating);
+            String title = rating.getGame();
+            String user = rating.getUser();
+            if (findByGameIdAndUserId(title, user) == null) {
                 em.persist(newRating);
-                System.out.println("###PERSISTED###");
+                System.out.printf("LOG: User %s added a new rating of game '%s' rated at %d",user,title,rating.getRating());
             }else {
-                em.createQuery("UPDATE Rating r SET r.rating = :rating WHERE r.game.title = :game")
+                em.createQuery("UPDATE Rating r SET r.rating = :rating, r.creationDate = :date WHERE r.game = :game AND r.user = :user")
                         .setParameter("rating", rating.getRating())
-                        //.setParameter("date", rating.getCreationDate())
-                        //.setParameter("user", rating.getUser())
-                        .setParameter("game", rating.getGame())
-                        //.getSingleResult());
+                        .setParameter("date", rating.getCreationDate())
+                        .setParameter("user", newRating.getUser())
+                        .setParameter("game", newRating.getGame())
                         .executeUpdate();
-                System.out.println("###UPDATED RATING###");
+                System.out.printf("LOG: User %s changed their rating of game '%s' to %d",user,title,rating.getRating());
             }
+            em.createQuery("UPDATE Game g SET g.averageRating = :average WHERE g.title = :game")
+                    .setParameter("average",getAverageOfGame(rating.getGame()))
+                    .setParameter("game", rating.getGame())
+                    .executeUpdate();
         } catch (Exception e){
             System.out.println("Problem writing to database : " + e );
         }
-        em.createQuery("UPDATE Game g SET g.averageRating = :average WHERE g.title = :game")
-                .setParameter("average",getAverageOfGame(rating.getGame()))
-                .setParameter("game", rating.getGame())
-                .executeUpdate();
-        System.out.println("###UPDATED AVERAGE###");
     }
 
-    private Rating ratingModelToEntity(RatingModel model) {
+    private Rating ratingModelToEntity(RatingModel model) throws Exception{
         Rating entity = new Rating();
-        entity.setGame(getGameFromTitle(model.getGame()));
-        entity.setUser(getUserFromUsername(model.getUser()));
+        try {
+            entity.setGame(getGameFromTitle(model.getGame()));
+            entity.setUser(getUserFromUsername(model.getUser()));
+        }catch (Exception e) {
+            throw new Exception("Failed to set entities: " + e.getMessage(), e);
+        }
```

Comment:
Be more specific with exception throwing and catching if possible. Also, it's pretty pointless to catch an exception just to throw the exact same one again.